src/caustics/sims/lens_source.py

```diff
@@ -229,75 +330,63 @@ def forward(
         psf_convolve: boolean
             when true the image will be convolved with the psf
         """
-        z_s, x0, y0 = self.unpack(params)
+        z_s, psf, x0, y0 = self.unpack(params)
 
         # Automatically turn off light for missing objects
         if self.source is None:
             source_light = False
         if self.lens_light is None:
             lens_light = False
-        if self.psf is None:
+        if psf.shape == (1, 1):
```

Comment:
If we use a kernel of 1 means we always convolve with the PSF, do we still need this?

Reply:
I think this is still useful. Convolving with a 1x1 kernel using the default fft method involves an fft and inverse fft. This unecessarily limits the dynamic range for a null operation. So just adding a little check to skip it saves someone else a headache.

Reply:
Yeah just realized there is the fft, though couldn’t we do something similar to the Pixelated Convergence where we decide which method to use at init or with the property trigger?

Reply:
Basically we would only use fft if the shape of the PSF exceed something like 32x32

Reply:
I made it so the user can choose the method with psf_mode. I feel like an automatic selector might cause unexpected behavior